src/scene/asset.cpp

```diff
@@ -475,408 +218,124 @@ void Asset::unload() {
 
     // 1. Initialize requirements
-    for (const std::shared_ptr<Asset>& child : _requiredAssets) {
+    for (Asset* child : _requiredAssets) {
         child->initialize();
     }
 
-    // 2. Initialize requests
-    for (const std::shared_ptr<Asset>& child : _requestedAssets) {
-        if (child->isSynchronized()) {
-            child->initialize();
-        }
-    }
-
-    // 3. Call lua onInitialize
+    // 2. Call Lua onInitialize
     try {
-        loader()->callOnInitialize(this);
+        _manager.callOnInitialize(this);
     }
     catch (const ghoul::lua::LuaRuntimeException& e) {
-        LERROR(fmt::format("Failed to initialize asset {}", id()));
+        LERROR(fmt::format("Failed to initialize asset {}", path()));
         LERROR(fmt::format("{}: {}", e.component, e.message));
-        // TODO: rollback;
         setState(State::InitializationFailed);
-        return false;
+        return;
     }
 
-    // 4. Update state
+    // 3. Update state
     setState(State::Initialized);
-
-    // 5. Call dependency lua onInitialize of this and requirements
-    for (const std::shared_ptr<Asset>& child : _requiredAssets) {
-        try {
-            loader()->callOnDependencyInitialize(child.get(), this);
-        }
-        catch (const ghoul::lua::LuaRuntimeException& e) {
-            LERROR(fmt::format(
-                "Failed to initialize required asset {} of {}; {}: {}",
-                child->id(), id(), e.component, e.message
-            ));
-            // TODO: rollback;
-            setState(State::InitializationFailed);
-            return false;
-        }
-    }
-
-    // 6. Call dependency lua onInitialize of this and initialized requests
-    for (const std::shared_ptr<Asset>& child : _requestedAssets) {
-        if (child->isInitialized()) {
-            try {
-                loader()->callOnDependencyInitialize(child.get(), this);
-            }
-            catch (const ghoul::lua::LuaRuntimeException& e) {
-                LERROR(fmt::format(
-                    "Failed to initialize requested asset {} of {}; {}: {}",
-                    child->id(), id(), e.component, e.message
-                ));
-                // TODO: rollback;
-            }
-        }
-    }
-
-    // 7. Call dependency lua onInitialize of initialized requesting assets and this
-    for (const std::weak_ptr<Asset>& parent : _requestingAssets) {
-        std::shared_ptr<Asset> p = parent.lock();
-        if (p && p->isInitialized()) {
-            try {
-                loader()->callOnDependencyInitialize(this, p.get());
-            }
-            catch (const ghoul::lua::LuaRuntimeException& e) {
-                LERROR(fmt::format(
-                    "Failed to initialize required asset {} of {}; {}: {}",
-                    id(), p->id(), e.component, e.message
-                ));
-                // TODO: rollback;
-            }
-        }
-    }
-    return true;
 }
 
+/*
```

Comment:
Is there a reason that this function is kept around even though it's commented out? If so, we need a comment explaining why

Reply:
No reason;  just forgot to actually remove it after commenting it out